File: internal/dto/credit.go
package dto

type CreateCreditRequest struct {
	AccountID       int64  `json:"account_id"`
	PrincipalAmount string `json:"principal_amount"`
	TermMonths      int    `json:"term_months"`
	MFACode         string `json:"mfa_code"`
}

type CheckCreditRequest struct {
	AccountID       int64  `json:"account_id"`
	PrincipalAmount string `json:"principal_amount"`
	TermMonths      int    `json:"term_months"`
}

type CreditPrepaymentRequest struct {
	Amount  string `json:"amount"`
	Mode    string `json:"mode"`
	MFACode string `json:"mfa_code"`
}

type CreditPrepaymentResponse struct {
	TransactionID     int64          `json:"transaction_id"`
	Credit            CreditResponse `json:"credit"`
	Amount            string         `json:"amount"`
	Mode              string         `json:"mode"`
	OldMonthlyPayment string         `json:"old_monthly_payment"`
	NewMonthlyPayment string         `json:"new_monthly_payment"`
	OldTermMonths     int            `json:"old_term_months"`
	NewTermMonths     int            `json:"new_term_months"`
	RemainingDebt     string         `json:"remaining_debt"`
	Closed            bool           `json:"closed"`
}

type CreditCheckResponse struct {
	Eligible                  bool     `json:"eligible"`
	Reason                    string   `json:"reason,omitempty"`
	Reasons                   []string `json:"reasons,omitempty"`
	PolicyEnabled             bool     `json:"policy_enabled"`
	AccountID                 int64    `json:"account_id"`
	PrincipalAmount           string   `json:"principal_amount"`
	MaxPrincipalAmount        string   `json:"max_principal_amount"`
	TermMonths                int      `json:"term_months"`
	InterestRate              string   `json:"interest_rate"`
	MonthlyPayment            string   `json:"monthly_payment"`
	ActiveCreditsCount        int      `json:"active_credits_count"`
	MaxActiveCredits          int      `json:"max_active_credits"`
	HasOverdueCredit          bool     `json:"has_overdue_credit"`
	TotalPrincipalAmount      string   `json:"total_principal_amount"`
	MaxTotalPrincipalAmount   string   `json:"max_total_principal_amount"`
	MonthlyIncome             string   `json:"monthly_income"`
	CurrentMonthlyPayments    string   `json:"current_monthly_payments"`
	RequestedMonthlyPayment   string   `json:"requested_monthly_payment"`
	TotalMonthlyPayments      string   `json:"total_monthly_payments"`
	MaxAllowedMonthlyPayments string   `json:"max_allowed_monthly_payments"`
	DebtLoadPercent           string   `json:"debt_load_percent"`
	MaxDebtLoadPercent        int      `json:"max_debt_load_percent"`
}

type CreditResponse struct {
	ID              int64  `json:"id"`
	AccountID       int64  `json:"account_id"`
	PrincipalAmount string `json:"principal_amount"`
	InterestRate    string `json:"interest_rate"`
	TermMonths      int    `json:"term_months"`
	MonthlyPayment  string `json:"monthly_payment"`
	Status          string `json:"status"`
	CreatedAt       string `json:"created_at"`
}

type PaymentScheduleResponse struct {
	ID            int64  `json:"id"`
	CreditID      int64  `json:"credit_id"`
	PaymentDate   string `json:"payment_date"`
	Amount        string `json:"amount"`
	PenaltyAmount string `json:"penalty_amount"`
	Status        string `json:"status"`
	PaidAt        string `json:"paid_at,omitempty"`
}

type CreditOperationResponse struct {
	Source        string `json:"source"`
	EventType     string `json:"event_type"`
	TransactionID *int64 `json:"transaction_id,omitempty"`
	ScheduleID    *int64 `json:"schedule_id,omitempty"`
	Amount        string `json:"amount,omitempty"`
	PenaltyAmount string `json:"penalty_amount,omitempty"`
	Status        string `json:"status,omitempty"`
	Description   string `json:"description,omitempty"`
	PaymentDate   string `json:"payment_date,omitempty"`
	OccurredAt    string `json:"occurred_at,omitempty"`
}

File: web/src/types/credit.ts
export type CreditCheckResponse = {
  eligible: boolean
  reason?: string
  reasons?: string[]
  policy_enabled: boolean
  account_id: number
  principal_amount: string
  max_principal_amount: string
  term_months: number
  interest_rate: string
  monthly_payment: string
  active_credits_count: number
  max_active_credits: number
  has_overdue_credit: boolean
  total_principal_amount: string
  max_total_principal_amount: string
  monthly_income: string
  current_monthly_payments: string
  requested_monthly_payment: string
  total_monthly_payments: string
  max_allowed_monthly_payments: string
  debt_load_percent: string
  max_debt_load_percent: number
}

export type CreditResponse = {
  id: number
  account_id: number
  principal_amount: string
  interest_rate: string
  term_months: number
  monthly_payment: string
  status: string
  created_at: string
}

export type CreditPrepaymentMode = 'reduce_payment' | 'reduce_term' | 'full_close'

export type CreditPrepaymentResponse = {
  transaction_id: number
  credit: CreditResponse
  amount: string
  mode: CreditPrepaymentMode
  old_monthly_payment: string
  new_monthly_payment: string
  old_term_months: number
  new_term_months: number
  remaining_debt: string
  closed: boolean
}

export type PaymentScheduleResponse = {
  id: number
  credit_id: number
  payment_date: string
  amount: string
  penalty_amount: string
  status: string
  paid_at?: string
}
